bot/cogs/news.py

```diff
@@ -0,0 +1,228 @@
+
+            for thread in recents.html.body.div.find_all("a", href=True):
+                # We want only these threads that have identifiers
+                if "latest" in thread["href"]:
+                    continue
+
+                thread_information, email_information = await self.get_thread_and_first_mail(
+                    maillist, thread["href"].split("/")[-2]
+                )
+
+                try:
+                    new_date = datetime.strptime(email_information["date"], "%Y-%m-%dT%X%z")
+                except ValueError:
+                    log.warning(f"Invalid datetime from Thread email: {email_information['date']}")
+                    continue
+
+                if (
+                        await self.check_new_exist(thread_information["subject"], new_date, maillist, existing_news)
+                        or new_date.date() < date.today()
+                ):
+                    continue
+
+                content = email_information["content"]
+                link = THREAD_URL.format(id=thread["href"].split("/")[-2], list=maillist)
+                msg_id = await self.send_webhook(
+                    title=thread_information["subject"],
+                    description=content[:500] + f"... [continue reading]({link})" if len(content) > 500 else content,
+                    timestamp=new_date,
+                    url=link,
+                    author=f"{email_information['sender_name']} ({email_information['sender']['address']})",
+                    author_url=MAILMAN_PROFILE_URL.format(id=email_information["sender"]["mailman_id"]),
+                    webhook_profile_name=self.webhook_names[maillist]
+                )
+                payload["data"][maillist].append(msg_id)
+
+        await self.bot.api_client.put("bot/bot-settings/news", json=payload)
+
+    async def check_new_exist(self, title: str, timestamp: datetime, maillist: str, news: t.Dict[str, t.Any]) -> bool:
+        """Check does this new title + timestamp already exist in #python-news."""
+        for new in news["data"][maillist]:
+            message = discord.utils.get(self.bot.cached_messages, id=new)
+            if message is None:
+                message = await self.channel.fetch_message(new)
+                if message is None:
+                    return False
+
+            if message.embeds[0].title == title and message.embeds[0].timestamp == timestamp:
+                return True
+        return False
+
+    async def send_webhook(self,
+                           title: str,
+                           description: str,
+                           timestamp: datetime,
+                           url: str,
+                           webhook_profile_name: str,
+                           author: t.Optional[str] = None,
+                           author_url: t.Optional[str] = None,
+                           ) -> int:
+        """Send webhook entry and return ID of message."""
+        embed = discord.Embed(
```

Comment:
This embed should have a footer. We intend to have the webhook be relayed to subscribing servers and we lose author data when that happens.

I propose an embed structure similar to:

![image](https://cdn.discordapp.com/attachments/635950537262759947/704086044911599676/unknown.png)
![image](https://user-images.githubusercontent.com/20439493/80320718-f84e7600-880f-11ea-95b1-7e35493c5a50.png)

This maintains the context when a message is crossposted, a similar footer should be used for PEPs.